cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MigrationTableUtility.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.metrics.process;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.data2.dataset2.lib.table.MetricsTable;
+import co.cask.cdap.data2.util.TableId;
+import co.cask.cdap.data2.util.hbase.HBaseDDLExecutorFactory;
+import co.cask.cdap.data2.util.hbase.HBaseTableUtil;
+import co.cask.cdap.metrics.store.MetricDatasetFactory;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.spi.hbase.HBaseDDLExecutor;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.client.HBaseAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import javax.annotation.Nullable;
+
+/**
+ * To get v2/v3 metrics tables; perform tableExists/delete table on v2 metrics tables
+ */
+public class MigrationTableUtility {
+  private static final Logger LOG = LoggerFactory.getLogger(MigrationTableUtility.class);
+
+  private final CConfiguration cConf;
+  private final Configuration hConf;
+  private final MetricDatasetFactory metricDatasetFactory;
+  private final HBaseDDLExecutorFactory hBaseDDLExecutorFactory;
+  private final HBaseTableUtil tableUtil;
+
+  MigrationTableUtility(CConfiguration cConfiguration, Configuration hConf,
+                        MetricDatasetFactory metricDatasetFactory, HBaseTableUtil tableUtil) {
+    this.cConf = cConfiguration;
+    this.hConf = hConf;
+    this.metricDatasetFactory = metricDatasetFactory;
+    this.hBaseDDLExecutorFactory = new HBaseDDLExecutorFactory(cConf, hConf);
+    this.tableUtil = tableUtil;
+  }
+
+  // we try to get v2 metrics table; if there is any exception while getting the table we return null
+  @Nullable
+  MetricsTable getV2MetricsTable(int resolution) {
+    MetricsTable v2table = null;
+    try {
+      v2table = metricDatasetFactory.getV2MetricsTable(resolution);
+    } catch (Exception e) {
```

Comment:
any exception? What if it is an OutOfMemory? Or IllegalArgumentException? Isn't there a (set of) expected exception(s) if the system is still starting up?